src/inmanta/export.py

```diff
@@ -438,7 +441,32 @@ def deploy_code(self, conn: protocol.SyncClient, tid: uuid.UUID, version: Option
 
         upload_code(conn, tid, version, code_manager)
 
-    def commit_resources(self, version: int, resources: List[Dict[str, str]], metadata: Dict[str, str], model: Dict) -> None:
+    def get_resource_sets(self, resource_set_instances: Optional[List[Instance]]) -> Optional[Dict[str, Optional[str]]]:
+        """
+        return a dictonary with as keys resource_ids and as values the name of the resource_set
+        the resource belongs to. return None if no resource_set is defined.
+        """
+        resource_sets: Optional[Dict[str, Optional[str]]] = {} if resource_set_instances else None
+        for resource_set_instance in resource_set_instances:
+            name = resource_set_instance.get_attribute("name").value
+            resources_in_set = resource_set_instance.get_attribute("resources")
+            for resource_in_set in resources_in_set.value:
+                resource_id = str(
+                    Resource.create_from_model(self, str(resource_in_set.type), DynamicProxy.return_value(resource_in_set))
+                )
+                if resource_id in resource_sets:
+                    raise CompilerException("resource '%s' can not be part of multiple ResourceSets" % resource_id)
```

Comment:
I'm not sure this should be a compilerException?

Reply:
I think this is OK